lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs

```diff
@@ -168,11 +170,54 @@ simpleSpec = do
             runExceptT (putCheckpoint db testPk testCp) `shouldReturn` Right ()
             readCheckpoint db testPk `shouldReturn` Just testCp
 
+loggingSpec :: Spec
+loggingSpec = withLoggingDB $ do
+    describe "Sqlite query logging" $ do
+        it "should log queries at DEBUG level" $ \(db, getLogs) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            logs <- logMessages <$> getLogs
+            logs `shouldHaveLog` (Debug, "INSERT")
+
+        it "should not log query parameters" $ \(db, getLogs) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            let walletName = T.unpack $ coerce $ name testMetadata
+            msgs <- T.unlines . map snd . logMessages <$> getLogs
+            T.unpack msgs `shouldNotContain` walletName
+
+    describe "Sqlite observables" $ do
+        it "should measure query timings" $ \(db, getLogs) -> do
+            replicateM_ 5 $ listWallets db
+            msgs <- T.unlines . map snd . logMessages <$> getLogs
+            T.unpack msgs `shouldContain` "runQuery monitor works"
 
 newMemoryDBLayer :: IO (DBLayer IO (SeqState DummyTarget) DummyTarget, TVar [LogObject Text])
 newMemoryDBLayer = do
+    logConfig <- defaultConfigTesting
+    CM.setMinSeverity logConfig Debug
+    CM.setSetupBackends logConfig [CM.KatipBK, CM.AggregationBK, CM.MonitoringBK]
+
+    CM.setBackends logConfig
+        "query.runQuery"
+        (Just [CM.AggregationBK])
+    CM.setAggregatedKind logConfig
+        "query.runQuery"
+        (Just CM.StatsAK) -- statistics AgreggatedKind
+    CM.setBackends logConfig
+        "#aggregation.query.runQuery"
+        (Just [CM.KatipBK])
+    CM.setMonitors logConfig $ HM.singleton
+        "query.runQuery"
+        ( Just (Compare "ObserveDiff" (GE, (OpMeasurable 0)))
+        , Compare "ObserveDiff" (GE, (OpMeasurable 0))
+        , [CreateMessage Info "runQuery monitor works"]
+        )
+
+    CM.setBackends logConfig
+        "query.runQuery"
```

Comment:
I wonder if setting backends somehow overlaps with setting monitors and that is why we do not see `runQuery monitor works`

Reply:
Or maybe as we have `Debug` severity and setting monitors with `CreateMessage Info` is having wrong log severity ....